### PR TITLE
replacing insufficient VST3 SDK conversion with proper version.

### PR DESCRIPTION
**Pull request by defiantnerd**
see issue https://github.com/free-audio/clap-wrapper/issues/305

### CODE REVIEW

File: src/wrapasvst3.cpp [hunk 1]
```diff
@@ -42,6 +42,82 @@ struct ClapHostExtensions
...
 #endif
 
+void utf8_to_utf16l(const char* utf8string, uint16_t* target, size_t targetsize)
+{
+  uint32_t codepoint = 0;
+  size_t targetpos = 0;
+
+  auto src = reinterpret_cast<const uint8_t*>(utf8string);
+  size_t pos = 0;
+  while (src[pos] && (targetpos < (targetsize - 1)))
```
Comment (defiantnerd) [inline src/wrapasvst3.cpp:52]: `(targetsize -2)` because a character and NUL must have room.

**Review by baconpaul — APPROVED** [review]
This all looks fine to me. The only question is are we sure `Steinberg::char16` and `uint16_t` are always the same type.

I would be tempted to add somewhere

```
static_assert(std::is_same_v<Steinberg::char16, uint16_t>);
```

just in case an API shifts underneath us. But glad to hear why we don't need that too!

**Comment by lewloiwc** [comment]
The parameter names were displayed correctly when I tried building with the code here!! Please continue with the parameter values as well! Once again, thank you all for working on this problem!

**Comment by defiantnerd** [comment]
Just checked in that values and bus names are also correctly converted.

**Pull request merged**